src/api/app/models/staging_workflow.rb

```diff
@@ -12,4 +12,8 @@ def unassigned_requests
   def ignored_requests
     BsRequest.none # TODO: define this method
   end
+
+  def free_staging_projects
+    staging_projects.where.not(id: staged_requests.pluck(:staging_project_id))
```

Comment:
1) This should be defined / a scope on the relationship
2) This is not very efficient 

This is untested but something similar should work:
```
   has_many :staging_projects, class_name: 'Project', inverse_of: :staging_workflow, dependent: :nullify do 
      scope :empty, joins(:requests).where(request: { request_id: nil })
   end
```

Reply:
@ChrisBr:

> 1) This should be defined / a scope on the relationship
> 2) This is not very efficient 
> 
> This is untested but something similar should work:
> ```
>    has_many :staging_projects, class_name: 'Project', inverse_of: :staging_workflow, dependent: :nullify do 
>       scope :empty, joins(:requests).where(request: { request_id: nil })
>    end
> ```

The code was moved to a method inside the relationship, called `without_staged_requests`. The query is improved with your suggestion. Please re-review.